fix compiler error on x86
inspired by uclibc-0002-Fix-__libc_epoll_pwait-compile-failure-on-x86.patch from the buildroot project
--- libc/sysdeps/linux/common/fallocate64.c
+++ libc/sysdeps/linux/common/fallocate64.c
@@ -26,8 +26,12 @@
 {
 	int ret;
 	INTERNAL_SYSCALL_DECL(err);
+	int offset_hi = offset >> 32;
+	int offset_lo = offset & 0xffffffff;
+	int len_hi    = len    >> 32;
+	int len_lo    = len    & 0xffffffff;
 	ret = (int) (INTERNAL_SYSCALL(fallocate, err, 6, fd, mode,
-		OFF64_HI_LO (offset), OFF64_HI_LO (len)));
+		offset_hi, offset_lo,len_hi, len_lo));
 	if (unlikely(INTERNAL_SYSCALL_ERROR_P (ret, err))) {
 		__set_errno(INTERNAL_SYSCALL_ERRNO (ret, err));
 		ret = -1;
